src/LaunchDarkly.EventSource/EventSource.cs

```diff
@@ -354,43 +361,85 @@ private void SetReadyState(ReadyState state, Action<StateChangedEventArgs> actio
             }
         }
 
-        private void ProcessField(string field, string value)
+        private void HandleParsedLine(EventParser.Result result)
         {
-            if (EventParser.IsDataFieldName(field))
+            if (result.IsComment)
             {
-                _eventBuffer.Add(value);
-                _eventBuffer.Add("\n");
+                OnCommentReceived(new CommentReceivedEventArgs(result.GetValueAsString()));
             }
-            else if (EventParser.IsIdFieldName(field))
+            else if (result.IsDataField)
             {
-                _lastEventId = value;
+                if (result.ValueString != null)
+                {
+                    if (_eventDataStringBuffer is null)
+                    {
+                        _eventDataStringBuffer = new List<string>(2);
+                    }
+                    _eventDataStringBuffer.Add(result.ValueString);
+                    _eventDataStringBuffer.Add("\n");
```

Comment:
It may seem a bit odd to use an array of strings rather than for instance a `StringBuilder`, but the previous author was actually being clever in this case (in the previous version of this same logic on lines 361-362). It's likely that most events have only a single line of `data:`. In that case, after allocating the initial `List<string>`, it'll simply take the `string` instance for the line that it's just parsed and put that in as element 0, and then put the literal `"\n"` in as element 1; then, when it realizes there are no more lines, it'll call `string.Concat` on just the first element, which just returns the existing `string`. And then the same `List` will be reused, so it's avoiding unnecessary allocations which would happen if we were appending to a buffer.